README.md

```diff
@@ -71,6 +64,367 @@ Connected (press CTRL+C to quit)
+- `PORT`: which port the server should listen to (default `3005`)
+
+### API
+
+As its name suggests, `blockfrost-websocket-link` offers a web socket based API: the client needs to connect to it
+through a web socket. Once the web socket connection is correctly established the server is immediately ready to accept
+commands. The communication (both ways) is _JSON encoded_: the server expects JSON encoded messages and will respond
+using the same encoding.
+
+Each input message (also known as command) requires a nonce which is mirrored by the server in the output message; it
+is not actually used by the server but it useful for the client to reconcile output messages to relative input message.
+
+Each input message requires a command as well (details will follow) and optionally a set of parameters, depending on
+the command.
+
+The payload of an input message is:
+
+```
+{
+  "id": ...,        // string or number nonce
+  "command": "...", // command
+  "params": {       // optional
+    ...             // varying parameters based on command
+  }
+}
+```
+
+Commands are divided in two main categories:
+- _request / response_ commands: each one of them originates one response output message;
+- _subscription_ commands: each one of them originates one immediate response output message with a feedback about the
+  requested operation; later, in case of successful subscription, more output messages will be generated based on the
+  event the command subscribed.
+
+The payload of an output message of a successful action is:
+
+```
+{
+  "id": ...,   // nonce
+  "data": ..., // message data
+  "type": "message"
+}
+```
+
+The payload of an output message of a unsuccessful action is:
+
+```
+{
+  "id": ..., // nonce
+  "data": {
+    "error": {
+      ...    // error details
+    }
+  },
+  "type": "error"
+}
+```
+
+#### Request / response commands
+
+- [`ESTIMATE_FEE`](#estimate_fee) - estimate transaction submission fee
+- [`GET_ACCOUNT_INFO`](#get_account_info) - get general information about an account
+- [`GET_ACCOUNT_UTXO`](#get_account_utxo) - get unspent UTxOs of an account
+- [`GET_BALANCE_HISTORY`](#get_balance_history) - get balance history of an account
+- [`GET_BLOCK`](#get_block) - get details of a block
+- [`GET_TRANSACTION`](#get_transaction) - get details of a transaction
+- [`GET_SERVER_INFO`](#get_server_info) - get information about the server
+- [`PUSH_TRANSACTION`](#push_transaction) - submits a transaction to the network
+- [`UNSUBSCRIBE_ADDRESS`](#unsubscribe_address) - cancel all new transactions subscriptions
+- [`UNSUBSCRIBE_BLOCK`](#unsubscribe_block) - cancel new blocks subscription
+
+#### Subscription commands
+
+- [`SUBSCRIBE_ADDRESS`](#subscribe_addresses) - subscribe to new transactions for given addresses
+- [`SUBSCRIBE_BLOCK`](#subscribe_block) - subscribe to new blocks
+
+#### ESTIMATE_FEE
+
+Input message:
+
+```
+{
+  "id": ..., // nonce
+  "command": 'ESTIMATE_FEE'
+}
+```
+
+TODO
```

Comment:
We could paste here TS type of the response? Or just raw example of the response? What do you think would be the best?
```ts
{
  "id": string | number,
  "command": 'ESTIMATE_FEE',
  "data": { lovelacePerByte: number },
}
```
https://github.com/blockfrost/blockfrost-websocket-link/blob/master/src/methods/estimate-fee.ts

Reply:
be54e33 should solve this